compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -1190,7 +1188,13 @@ object RefChecks {
             }
           }
         .exists
-      if !target.typeSymbol.isOpaqueAlias && hidden
+      val receiverName = sym.info.explicit.firstParamNames.head
+      val num = sym.info.paramNamess.flatten.indexWhere(_ == receiverName)
+      val getterName = DefaultGetterName(sym.name.toTermName, num = num)
+      val getterDenot = sym.owner.info.member(getterName)
```

Comment:
nit: Can you move that into a local function? The logic might be slightly clearer then.

Reply:
also leverage `HasDefaultParams`